hikari/files.py

```diff
@@ -757,142 +754,63 @@ def filename(self) -> str:
-
-
-def _stat(path: pathlib.Path) -> pathlib.Path:
-    # While paths will be implicitly resolved, we still need to explicitly
-    # call expanduser to deal with a ~ base.
-    try:
-        path = path.expanduser()
-    except RuntimeError:
-        pass  # A home directory couldn't be resolved, so we'll just use the path as-is.
-
-    # path.stat() will raise FileNotFoundError if the file doesn't exist
-    # (unlike is_dir) which is what we want here.
-    if stat.S_ISDIR(path.stat().st_mode):
-        raise IsADirectoryError(errno.EISDIR, "Cannot open the path specified as it is a directory", str(path))
-
-    return path
-
-
-@attr.define(weakref_slot=False)
-@typing.final
-class _ThreadedFileReaderContextManagerImpl(AsyncReaderContextManager[FileReader]):
-    executor: typing.Optional[concurrent.futures.ThreadPoolExecutor] = attr.field()
-    file: typing.Optional[typing.BinaryIO] = attr.field(default=None, init=False)
-    filename: str = attr.field()
-    path: pathlib.Path = attr.field()
-
-    async def __aenter__(self) -> ThreadedFileReader:
-        if self.file:
-            raise RuntimeError("File is already open")
-
-        loop = asyncio.get_running_loop()
-        self.path = await loop.run_in_executor(self.executor, _stat, self.path)
-        self.file = typing.cast(io.BufferedReader, await loop.run_in_executor(self.executor, self.path.open, "rb"))
-        return ThreadedFileReader(self.filename, None, self.executor, self.path, self.file)
-
-    async def __aexit__(
-        self,
-        exc_type: typing.Optional[typing.Type[BaseException]],
-        exc: typing.Optional[BaseException],
-        exc_tb: typing.Optional[types.TracebackType],
-    ) -> None:
-        if not self.file:
-            raise RuntimeError("File isn't open")
-
-        loop = asyncio.get_running_loop()
-        file = self.file
-        self.file = None
-        await loop.run_in_executor(self.executor, file.close)
-
-
-@attr.define(weakref_slot=False)
-class ThreadedFileReader(FileReader):
+class ThreadedFileReader(AsyncReader):
     """Asynchronous file reader that reads a resource from local storage.
 
     This implementation works with pools that exist in the same interpreter
     instance as the caller, namely thread pool executors, where objects
     do not need to be pickled to be communicated.
     """
 
+    _executor: typing.Optional[concurrent.futures.ThreadPoolExecutor] = attr.field()
     _pointer: typing.BinaryIO = attr.field()
 
     async def __aiter__(self) -> typing.AsyncGenerator[typing.Any, bytes]:
         loop = asyncio.get_running_loop()
 
         while True:
-            chunk = await loop.run_in_executor(self.executor, self._pointer.read, _MAGIC)
+            chunk = await loop.run_in_executor(self._executor, self._pointer.read, _MAGIC)
             yield chunk
             if len(chunk) < _MAGIC:
                 break
 
 
+def _open_path(path: pathlib.Path) -> typing.BinaryIO:
+    return path.expanduser().open("rb")
```

Comment:
Note for reviewers, `expanduser` will never raise a RuntimeError as stated in the docs and `open` will take care of opening directories with a `IsADirectory` error and unknown files with `FileNotFoundError`